File: app/src/main/java/com/hjc/base/model/LoginModel.java
package com.hjc.base.model;

import com.hjc.base.bean.LoginReq;
import com.hjc.base.bean.LoginResp;
import com.hjc.base.http.RetrofitClient;
import com.hjc.base.http.RxSchedulers;
import com.hjc.base.http.bean.BaseResponse;
import com.hjc.baselib.model.BaseModel;

import io.reactivex.Observable;

public class LoginModel extends BaseModel {

    public Observable<BaseResponse<LoginResp>> login(LoginReq loginReq) {
        return RetrofitClient.getInstance().getAPI()
                .login(loginReq)
                .compose(RxSchedulers.ioToMain());
    }
}

File: module-other/src/main/java/com/hjc/module_other/model/LoginModel.kt
package com.hjc.module_other.model

import com.hjc.library_common.model.CommonModel
import com.hjc.library_net.bean.BaseResponse
import com.hjc.library_net.utils.RxSchedulers
import com.hjc.module_other.http.OtherService
import com.hjc.module_other.http.entity.LoginBean
import io.reactivex.Observable

class LoginModel : CommonModel() {

    private val mApi = getApiService(OtherService::class.java)

    fun login(name: String?, password: String?): Observable<BaseResponse<LoginBean>> {
        return mApi.login(name, password)
            .compose(RxSchedulers.ioToMain())
    }
}